Paws4Me/Paws4Me/Show and Validate Login/View Models/SignInViewModel.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  ViewModels.swift
+//  Paws4Me
+//
+//  Created by Kirsty-Lee Walker on 2022/03/24.
+//
+
+import Foundation
+
+// MARK: - SignInViewModel Delegate
+protocol SignInViewModelDelegate: AnyObject {
+    func successRouting()
+    func show(errorMessage: String)
+}
+
+class SignInViewModel {
+
+    // MARK: - Vars/Lets
+    private weak var delegate: SignInViewModelDelegate?
+
+    init(delegate: SignInViewModelDelegate) {
+        self.delegate = delegate
+    }
+
+    // MARK: - Functions
+    func loginUser(username: String, password: String) {
+        if username.isEmpty || password.isEmpty {
+            delegate?.show(errorMessage: "Please fill in your username and password")
+        }
+        _ = isValidCredentials(username: username, password: password) ? delegate?.successRouting():
```

Comment:
No need for the `_ =` you can do this: 

`isValidCredentials(username: username, password: password) ? delegate?.successRouting() : delegate?.show(errorMessage: "Incorrect Username or Password")`

 Please also note spacing 😄

Reply:
Thank you 👍🏻